### PR TITLE
Cache dataset.pivot() and make cluster search box more visible

### PR DESCRIPTION
**Pull request by dsmilkov**
https://huggingface.co/spaces/lilacai/daniel_staging

### CODE REVIEW

**Review by brilee — APPROVED** [review]

File: lilac/data/dataset_duckdb.py [hunk 1]
```diff
@@ -1843,6 +1849,12 @@ def pivot(
     sort_order = sort_order or SortOrder.DESC
     inner_path = normalize_path(inner_path)
     outer_path = normalize_path(outer_path)
+
+    pivot_key = (outer_path, inner_path, sort_by, sort_order)
+    use_cache = not filters
+    if use_cache and pivot_key in self._pivot_cache:
```
Comment (nsthorat) [inline lilac/data/dataset_duckdb.py:1855]: what happens when data is deleted?

also, does the pivot query remove deleted rows (it should)

File: web/blueprint/src/lib/components/datasetView/DatasetPivotViewer.svelte [hunk 2]
```diff
@@ -162,15 +163,18 @@
 
 <div class="flex h-full flex-col">
   <div class="mb-8 flex h-16 w-full flex-row justify-between justify-items-center gap-x-4">
-    <div class="ml-8 mt-4 w-96">
-      <Search
+    <div class="ml-8 mt-4 flex w-96 items-center gap-x-2">
```
Comment (nsthorat) [inline web/blueprint/src/lib/components/datasetView/DatasetPivotViewer.svelte:166]: while you're here lets round the edges so it looks less IBM (and more like the searchbox and buttons above)

File: web/blueprint/src/lib/components/datasetView/DatasetPivotViewer.svelte [hunk 3]
```diff
@@ -162,15 +163,18 @@
...
         value={searchText}
         on:input={searchInput}
-        placeholder={`Search`}
-        labelText={'text text'}
+        placeholder="Search all categories and titles"
         on:change={search}
-        on:clear={clearSearch}
+        size="xl"
       />
+      <div use:hoverTooltip={{text: 'Clear search query'}}>
```
Comment (nsthorat) [inline web/blueprint/src/lib/components/datasetView/DatasetPivotViewer.svelte:175]: make this invisible / disabled when theres nothing in  the searchbox

**Review by nsthorat — APPROVED** [review]
Can we achieve consistency with the top search box:

image

Also notice how google aligns, the edge of the content aligns with the edge of the searchbox, so you get a straight edge with text:
image

Also, it's really weird that one of these icons is a button and the other isnt, can we make the search a button (and just move it to the right, like Google):
image

File: web/blueprint/src/lib/components/datasetView/DatasetPivotViewer.svelte (hunk 3 above)
Comment (dsmilkov) [reply web/blueprint/src/lib/components/datasetView/DatasetPivotViewer.svelte]: done

**Pull request merged**